apps/examples/testcase/le_tc/kernel/tc_memory_safety.c

```diff
@@ -0,0 +1,408 @@
+#else
+#define STACKSIZE 1024
+#endif
+
+static mqd_t g_send_mqfd;
+static mqd_t g_recv_mqfd;
+static int g_exit;
+
+/**
+* @fn                   :sender_thread
+* @description          :Function for tc_memory_safety
+* @return               :void*
+*/
+static void *sender_thread(void *arg)
+{
+	char msg_buffer[TEST_MSGLEN];
+	struct mq_attr attr;
+	int status = 0;
+	int nerrors = 0;
+	int msg_idx;
+
+	/* Fill in attributes for message queue */
+
+	attr.mq_maxmsg = 20;
+	attr.mq_msgsize = TEST_MSGLEN;
+	attr.mq_flags = 0;
+
+	/* Set the flags for the open of the queue.
+	 * Make it a blocking open on the queue, meaning it will block if
+	 * this process tries to send to the queue and the queue is full.
+	 *
+	 *   O_CREAT - the queue will get created if it does not already exist.
+	 *   O_WRONLY - we are only planning to write to the queue.
+	 *
+	 * Open the queue, and create it if the receiving process hasn't
+	 * already created it.
+	 */
+
+	g_send_mqfd = mq_open("mqueue", O_WRONLY | O_CREAT, 0666, &attr);
+	if (g_send_mqfd == (mqd_t)-1) {
+		printf("tc_mqueue_mq_open FAIL\n");
+		pthread_exit((pthread_addr_t)1);
+	}
+
+	/* Fill in a test message buffer to send */
+
+	memcpy(msg_buffer, TEST_MESSAGE, TEST_MSGLEN);
+
+	/* Perform the send TEST_SEND_NMSGS times */
+
+	for (msg_idx = 0; msg_idx < TEST_SEND_NMSGS; msg_idx++) {
+		status = mq_send(g_send_mqfd, msg_buffer, TEST_MSGLEN, 42);
+		if (status < 0) {
+			printf("tc_mqueue_mq_send FAIL : failure=%d on msg %d\n", status, msg_idx);
+			nerrors++;
+		}
+	}
+
+	/* Close the queue and return success */
+
+	if (mq_close(g_send_mqfd) < 0) {
+		printf("tc_mqueue_mq_close FAIL\n");
+	} else {
+		g_send_mqfd = NULL;
+	}
+
+	pthread_exit((pthread_addr_t)nerrors);
+	return (pthread_addr_t)nerrors;
+}
+
+/**
+* @fn                   :receiver_thread
+* @description          :Function for tc_memory_safety
+* @return               :void*
+*/
+static void *receiver_thread(void *arg)
+{
+	char msg_buffer[TEST_MSGLEN];
+	struct mq_attr attr;
+	int nbytes;
+	int nerrors = 0;
+	int msg_idx;
+
+	/* Fill in attributes for message queue */
+
+	attr.mq_maxmsg = 20;
+	attr.mq_msgsize = TEST_MSGLEN;
+	attr.mq_flags = 0;
+
+	/* Set the flags for the open of the queue.
+	 * Make it a blocking open on the queue, meaning it will block if
+	 * this task tries to read from the queue when the queue is empty
+	 *
+	 *   O_CREAT - the queue will get created if it does not already exist.
+	 *   O_RDONLY - we are only planning to read from the queue.
+	 *
+	 * Open the queue, and create it if the sending process hasn't
+	 * already created it.
+	 */
+
+	g_recv_mqfd = mq_open("mqueue", O_RDONLY | O_CREAT, 0666, &attr);
+	if (g_recv_mqfd == (mqd_t)ERROR) {
+		printf("tc_mqueue_mq_open FAIL\n");
+		pthread_exit((pthread_addr_t)1);
+	}
+
+	/* Perform the receive TEST_RECEIVE_NMSGS times */
+
+	for (msg_idx = 0; msg_idx < TEST_RECEIVE_NMSGS; msg_idx++) {
+		memset(msg_buffer, 0xaa, TEST_MSGLEN);
```

Comment:
Just a question, what is the meaning of "0xaa"?

Reply:
This might help discrimination between the original message and the received one.